services/st_util/FlashOnStBootloaderCommunicator.cpp

```diff
@@ -0,0 +1,69 @@
+#include "services/st_util/FlashOnStBootloaderCommunicator.hpp"
+#include "infra/util/ByteRange.hpp"
+#include "infra/util/MemoryRange.hpp"
+#include "services/st_util/StBootloaderCommunicator.hpp"
+#include <cstdint>
+
+namespace services
+{
+    FlashOnStBootloaderCommunicatorBase::FlashOnStBootloaderCommunicatorBase(hal::Flash& flash, StBootloaderCommunicator& communicator)
+        : communicator(communicator)
+        , flash(flash)
+        , onWriteMemoryDone([this]()
```

Comment:
These functions (onWriteMemoryDone, onReadMemoryDone, onExtendedEraseDone) are never reassigned. That means that they are actually normal functions in disguise. So either please add them as normal member functions, or just inline them at the location where they're used (since they're used only once, that would probably increase the readability)